docker/docker/src/main/java/org/arquillian/cube/docker/impl/model/DockerCube.java

```diff
@@ -139,10 +155,14 @@ public void stop() throws CubeControlException {
         }
         try {
             lifecycle.fire(new BeforeStop(id));
+
+            long currentTime = System.currentTimeMillis();
             try {
                 executor.stopContainer(id);
             } catch(NotFoundException e) {
             } catch (NotModifiedException e) {}
+            this.stoppingTimeInMillis = System.currentTimeMillis() - currentTime;
+
```

Comment:
Yes as I said I know it is not the best way in terms of code, but at least it is the best in terms of accuracy. I can try to put in higher priority the event which will not be the same times but really close.